src/coreclr/src/jit/optimizer.cpp

```diff
@@ -8659,6 +8659,145 @@ bool Compiler::optIsRangeCheckRemovable(GenTree* tree)
     return true;
 }
 
+//----------------------------------------------------------------------------------
+// optBranchlessConditions: optimize simple expressions to branchless expressions
+//
+//  Operation:
+//      Look for BBJ_COND basic blocks which target two BBJ_RETURN blocks
+//      and convert them into single branchless expressions, e.g.
+//
+//            JTRUE
+//              \
+//              EQ              RETURN
+//             /  \                \
+//            X    Y               ADD
+//                       =>       /   \
+//           RETURN             EQ     C3
+//              \              /  \
+//              C1            X    Y
+//
+//           RETURN
+//              \
+//               C2
+//
+void Compiler::optBranchlessConditions()
+{
+    JITDUMP("\n*************** In optBranchlessConditions()\n");
+    bool changed = false;
+    for (BasicBlock* block = fgFirstBB; block; block = block->bbNext)
+    {
+        if ((block->bbJumpKind != BBJ_COND) || block->isEmpty())
```

Comment:
Just curious why a check for `block->isEmpty()`. Don't we already remove empty blocks?